### ISSUE TITLE
Is `newspaper-order` too strict?

### ISSUE THREAD

**Issue by lukaselmer** [Glavin001/tslint-clean-code#71]
When I have the following code:

```ts
export function a() {
  console.log("a");
}

export function b() {
  c();
}

function c() {
  console.log("c");
}
```

`newspaper-order` tells me that a() needs to move to the bottom. Is this behavior intended? IMO `newspaper-order` should allow the following orders: `[a, b, c]` and `[b, c, a]`.

When writing an API, I would like to be able to write it like so:

```ts
export function index() {
  /* simple, doesn't need an additional function */
}

export function show() {
  /* simple, doesn't need an additional function */
}

export function update() {
  tooComplexForOneFunction();
}

function tooComplexForOneFunction() {}
```

So because `update` needs an additional helper function, `newspaper-order` tells me to move `update` to the top, which doesn't make sense to me.

This issue is especially problematic when using the rule with react. E.g. here, it would ask me to move `componentDidMount` to the bottom of the class, which I really don't want to do:

```tsx
export class MyComponent extends Component<{}, {}> {
  componentDidMount() {
    console.log('mounted');
  }

  render() {
    return (<div>{this.renderChildA()} {this.renderChildB()}</div>);
  }

  private renderChildA() {
    return 'a';
  }

  private renderChildB() {
    return 'b';
  }
}
```

**Comment by Glavin001** [comment]
`newspaper-order` rule can definitely get better!
 
> newspaper-order tells me that a() needs to move to the bottom. Is this behavior intended? IMO newspaper-order should allow the following orders: [a, b, c] and [b, c, a].

I agree, those orders should be allowed, too.

It would be great if you could create a Pull Request adding to the existing tests: https://github.com/Glavin001/tslint-clean-code/blob/master/src/tests/NewspaperOrderRuleTests.ts

I think the applicable code, or at least where to start debugging, could be https://github.com/Glavin001/tslint-clean-code/blob/master/src/newspaperOrderRule.ts#L381-L414

Would love to see a Pull Request making `newspaper-order` rule more flexible and powerful! Thank you!

**Comment by lukaselmer** [comment]
> I agree, those orders should be allowed, too.

Great! I thought I'd check with you if the current behavior is intentional before attempting to fixing it. I'll create a PR soon to improve it 😊

**Comment by Glavin001** [comment]
Merged https://github.com/Glavin001/tslint-clean-code/pull/72

Published to v0.2.10 🚀
